packages/react-router/lib/hooks.tsx

```diff
@@ -697,7 +697,11 @@ export function _renderMatches(
       return (
         <RenderedRoute
           match={match}
-          routeContext={{ outlet, matches }}
+          routeContext={{
+            outlet,
+            matches,
+            isDataRoute: dataRouterState != null,
```

Comment:
In order to get route-aware "am I in a data router" we add `isDataRoute` to the `RouteContext`